Userland/Services/WindowServer/Window.cpp

```diff
@@ -516,33 +517,34 @@ void Window::handle_keydown_event(KeyEvent const& event)
     if (event.modifiers() == Mod_Alt && event.code_point() && m_menubar.has_menus()) {
         // When handling alt shortcuts, we only care about the key that has been pressed in addition
         // to alt, not the code point that has been mapped to alt+[key], so we have to look up the
-        // scancode directly from the "unmodified" character map.
+        // kernel map_entry_index value directly from the "unmodified" character map.
         auto character_map_or_error = Keyboard::CharacterMap::fetch_system_map();
         if (!character_map_or_error.is_error()) {
             auto& character_map = character_map_or_error.value();
 
-            // The lowest byte serves as our index into the character table.
-            auto index = event.scancode() & 0xff;
-            u32 character = to_ascii_lowercase(character_map.character_map_data().map[index]);
-
-            Menu* menu_to_open = nullptr;
-            m_menubar.for_each_menu([&](Menu& menu) {
-                if (to_ascii_lowercase(menu.alt_shortcut_character()) == character) {
-                    menu_to_open = &menu;
-                    return IterationDecision::Break;
+            auto index = event.map_entry_index();
+            if (index != 0xff) {
```

Comment:
Do we need this check? it seems that the kernel giving us a Keyboard event with index 255 is completely invalid anyway. should it be an assert or verify instead?

Reply:
well, index 0xFF could be used for special keys, like Ctrl.